### ISSUE TITLE
[Feature]: vllm-ascend新增适配gemma3系列模型以及使用指导

### ISSUE THREAD

**Issue by NSDie** [vllm-project/vllm-ascend#2868]
### 适配gemma3模型

- [x] AscendAttention支持sliding windows参数，目前只支持prefill no hit和decode only。
- [x] 支持gemma3 w8a8量化、动态量化。
- [x] 支持prefix cache特性

### Alternatives

<img width="1449" height="374" alt="Image" src="https://github.com/user-attachments/assets/728a8966-2521-401f-aa3b-d81da39ede57" />

修改如下代码：
1. `vllm_ascend/core/schedule_config.py` ：

删除多模态相关的限制
<img width="2161" height="436" alt="Image" src="https://github.com/user-attachments/assets/87d2f52b-19fb-41da-ad5a-a231df7d598f" />

2. `vllm_ascend/core/scheduler.py` ：

<img width="2171" height="210" alt="Image" src="https://github.com/user-attachments/assets/642dfe54-5cc1-400b-8e7f-29ef2418b03d" />

3. `vllm_ascend/attention/attention_v1.py`:
在 `AscendAttentionBackendImpl`类里添加新的变量：
```python
        self.key_cache = None
        self.value_cache = None
        self.swa_mask = None
        if self.sliding_window is not None:
            mask = torch.ones(2048, 2048, dtype=torch.bool)
            triu_mask = torch.triu(mask, diagonal=1).to("npu")
            tril_mask = torch.tril(mask, -self.sliding_window).to("npu")
            self.swa_mask = triu_mask + tril_mask
```

4. `vllm_ascend/attention/attention_v1.py` ：
在`_forward_prefill_no_cache`里新增分支
```python
        if self.sliding_window is not None and \
            attn_metadata.attn_mask.shape[0] > self.sliding_window:

            slen = query.shape[0]
            dim = query.shape[2]
            query = query.view(1, slen, self.num_heads, dim).contiguous()
            key = key.view(1, slen, self.num_kv_heads, dim).contiguous()
            value = value.view(1, slen, self.num_kv_heads, dim).contiguous()
            

            output, _ = torch_npu.npu_fused_infer_attention_score(
                query,
                key,
                value,
                num_heads=self.num_heads,
                num_key_value_heads=self.num_kv_heads,
                input_layout="BSND",
                pre_tokens=self.sliding_window,
                next_tokens=0,
                atten_mask=self.swa_mask,
                sparse_mode=4,
                scale=self.scale,
                actual_seq_lengths=attn_metadata.seq_lens,
                actual_seq_lengths_kv=attn_metadata.seq_lens)
            
            output = output.view(num_tokens, self.num_heads, self.head_size)
        else:
            torch_npu._npu_flash_attention(query=query,
                                           key=key,
                                           value=value,
                                           mask=mask,
                                           seq_len=attn_metadata.seq_lens,
                                           scale_value=self.scale,
                                           num_heads=self.num_heads,
                                           num_kv_heads=self.num_kv_heads,
                                           out=output)
```
5. `vllm_ascend/attention/attention_v1.py` ：
在`_forward_prefill_cache_hit`里修改算子：
```python
    def _forward_prefill_cache_hit(
        self,
        query: torch.Tensor,
        attn_metadata: AscendMetadata,
        output: Optional[torch.Tensor] = None,
    ) -> torch.Tensor:
        assert attn_metadata is not None
        assert attn_metadata.attn_mask is not None

        compress_mask = attn_metadata.attn_mask
        batch_size = attn_metadata.query_lens.shape[0]
        block_table = attn_metadata.block_tables[:batch_size, :]
        block_size = self.key_cache.shape[1]
        num_tokens = query.shape[0]
        query = query.view(batch_size, num_tokens, self.num_heads * self.head_size).contiguous()
        key = self.key_cache
        value = self.value_cache
        if self.key_cache is not None and self.value_cache is not None:
            block_size = self.key_cache.shape[1]
            key = self.key_cache.flatten(2, 3).contiguous()
            value = self.value_cache.flatten(2, 3).contiguous()
        
        if self.sliding_window is not None:
            mask = self.swa_mask
            sparse_mode = 4
        else:
            mask = None
            sparse_mode = 0

        output, _ =torch_npu.npu_fused_infer_attention_score(
            query,
            key,
            value,
            num_heads=self.num_heads,
            num_key_value_heads=self.num_kv_heads,
            input_layout="BSH",
            pre_tokens=self.sliding_window if self.sliding_window is not None else 2147483647,
            next_tokens=0 if self.sliding_window is not None else 2147483647,
            atten_mask=mask,
            sparse_mode=sparse_mode,
            scale=self.scale,
            block_table=block_table,
            block_size=block_size,
            actual_seq_lengths=attn_metadata.query_lens,
            actual_seq_lengths_kv=attn_metadata.seq_lens
        )
        output = output.view(num_tokens, self.num_heads, self.head_size)
        return output
```

**Comment by NSDie** [comment]
Due to the fused_infer_attention operator, the current vllm_ascend=0.13.0 version does not support gemm3. This issue is expected to be resolved in 26.Q1.
